CMSgov/design-system
# [WNMGDS-700] Autocomplete component not rendering

Autocomplete component is not rendering on some applications like eADP.

Found that the application does not support `child.type` with the resulting Autocomplete data not displaying. 
 
### Changed
-  Add a fallback conditional `child.type.displayName and child.type.name` checks for Autocomplete component
- Table component has similar `child.type` logic as the Autocomplete component so it does not work on eADP either. Update Table component with similar changes.

### Fixed
This fixes https://github.com/CMSgov/design-system/issues/877

### How to test
- Run the doc site locally with `yarn start`
- Check that `Autocomplete` and scrollable feature of `Table` component are working
- On eADP, the `Autocomplete` component and `Table` scrollable feature do not work correctly
- Clone the [eADP repo](https://github.com/CMSgov/eAPD) to local system and setup a test environment. Follow the instruction on [setup readme.md](https://github.com/CMSgov/eAPD/blob/master/README.md) or [setting up manually](https://github.com/CMSgov/eAPD/wiki/Development-Environment#manually)
- Select one of the eADP component (eg. Root.js) for testing:
  - import the `Autocomplete` components (eg. copy the autocomplete examples from CMSDS). Check that `Autocomplete` works correctly
  - import the `Table` components (eg. copy the table examples from CMSDS). Check that `Table` works correctly
